### ISSUE TITLE
Retrace server should not fail

### ISSUE THREAD

**Issue by mcatanzaro** [abrt/retrace-server#303]
Moving this from https://github.com/abrt/abrt/issues/1457. When reporting a crash, ABRT first tries to use the retrace server. This usually takes a long time (10-15 minutes) and almost always fails, resulting in a poor user experience. Then, once it fails, ABRT falls back to installing debuginfo locally and generating the backtrace locally. This installs a huge amount of debuginfo, but it's much faster and seems more reliable.

To demonstrate, I decided to create a fake crash, for reproducibility:

`$ killall -SEGV geary`

I ran this at 2:57 PM, then waited a minute or so until ABRT "noticed" and popped up its desktop notification (https://github.com/abrt/abrt/issues/1455). Then I started the ABRT report process. Until 3:15 PM, ABRT was "Preparing environment for backtrace generation," so let's say that took somewhere between 15-18 minutes. (I know this is happening on the retrace server, but the user doesn't know that, and probably just thinks that ABRT is slow.) At 3:15 PM, ABRT displayed this window:

![Screenshot from 2020-03-03 15-15-57](https://user-images.githubusercontent.com/1424966/75820512-15138380-5d62-11ea-911c-e40458156b6f.png)

Note that the dialog tells me to report the issue, but doesn't tell me where to report the issue to, nor does it provide any sort of error code or message that could be used for debugging the issue. Hence, this bug report is necessarily extremely generic.

When I click Close, ABRT starts the Generating Backtrace step (local backtrace), which, for the purposes of this bug report, I cancelled. Then I was able to manually copy the problem directory for analysis before ABRT deleted it. You can probably recreate your own problem directory by killing Geary yourself, but here is mine:

[problem.zip](https://github.com/abrt/retrace-server/files/4283678/problem.zip)

In my experience, the retrace step almost always fails for almost every crash, for a couple of years now.

**Comment by mcatanzaro** [comment]
> In my experience, the retrace step almost always fails for almost every crash, for a couple of years now.

I just repeated the report process for this crash and the retrace server failed a second time, so I think this is nice and reproducible.

**Comment by mcatanzaro** [comment]
I haven't seen the retrace server succeed in processing any backtrace for a long time. I think it's not able to ever succeed. (Are there any metrics to track success/fail?)

**Comment by BjoernDaase** [comment]
@ernestask Is "soon" still in the future or already in the past? Because I still get this "retrace failed" message every time trying to report a bug via abrt.

**Comment by ernestask** [comment]
Not sure, since I left the team before things could start moving.

@xsuchy might be able to give an update.

**Comment by mcatanzaro** [comment]
I decided to test this again, to verify whether it is still broken.

> To demonstrate, I decided to create a fake crash, for reproducibility:
> 
> `$ killall -SEGV geary`

Geary seems to have started ignoring SEGV (!!!), but it works when using with SIGABRT instead.

The good news is that ABRT did not fail after 15-18 minutes this time. The bad news is that it didn't succeed either. After 45 minutes of waiting, I decided that any reasonable user would have given up long ago, and stopped the bug report process:

![Screenshot from 2020-12-07 10-09-51](https://user-images.githubusercontent.com/1424966/101375207-bc36c580-3874-11eb-805b-7c28f7d67398.png)

So whether it would work or not if given indefinite time, I'm not sure, but this seems equivalent to a fail to me: an experience like this teaches users that ABRT is going to waste their time, and that they should not attempt to report bugs in the future. Anything longer than five minutes is too long.
